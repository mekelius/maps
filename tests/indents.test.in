let main =
    ;
    ;


// whitespace and comments ignored

    ;
    ;    ; // this shouldn't matter
    ;
        ;
        ;
        ;
                ;
                ;
                // indented comments as well
                
                ;
                        ; 
                
                        // this is an edge case
                        let test = 67
                                    / 87
                        ; 
        ;
        ;
        ;
                    ;
                    let x = 123
                    ;
        ;
        ;
        ;
            ;
            ;       
                        ;
            ;
    ;

let gnu_style = {
        let y = 3
    }

let invalid_indents_roll_up =
    ;
    ;
                        ;
                ;
        ;
    ;
