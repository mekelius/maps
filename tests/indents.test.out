token: reserved word let
token: identifier main
token: whitespace 
token: operator =
token: indent block start
token: indent block start
token: indent block start
token: indent block start
token: reserved word let
token: identifier test
token: whitespace 
token: operator =
token: whitespace 
token: numeric literal 67
token: indent block start
token: operator /
token: whitespace
token: numeric literal 87
token: indent block end 1
token: indent block end 2
token: indent block start
token: reserved word let
token: identifier x
token: whitespace 
token: operator =
token: whitespace 
token: numeric literal 123
token: ;
token: indent block end 1
token: indent block start
token: indent block start
token: indent block end 1
token: indent block end 2
token: indent block end 1
token: reserved word let
token: identifier invalid
token: whitespace 
token: operator =
token: indent block start
token: indent block start
syntax error at line 44: Mismatched indent
token: ;
token: indent block start
token: indent block end 1